test/integration/plugins/storpool/TestStorPoolTiers.py

```diff
@@ -0,0 +1,554 @@
+                    %s" % self.hypervisor)
+        return
+
+    def tearDown(self):
+        return
+
+    @attr(tags=["advanced", "advancedns", "smoke"], required_hardware="true")
+    def test_01_check_tags_on_deployed_vm_and_datadisk(self):
+        virtual_machine_tier1_tag = self.deploy_vm_and_check_tier_tag()
+        virtual_machine_tier1_tag.stop(self.apiclient, forced=True)
+
+    @attr(tags=["advanced", "advancedns", "smoke"], required_hardware="true")
+    def test_02_change_offering_on_attached_root_disk(self):
+        virtual_machine_tier1_tag = self.deploy_vm_and_check_tier_tag()
+
+        root_volume = list_volumes(self.apiclient, virtualmachineid=virtual_machine_tier1_tag.id, type="ROOT",
+                                   listall=True)
+        self.changeOfferingForVolume(root_volume[0].id, self.disk_offerings_tier2_tags.id, root_volume[0].size)
+        root_volume = list_volumes(self.apiclient, virtualmachineid=virtual_machine_tier1_tag.id, type="ROOT",
+                                   listall=True)
+        self.vc_policy_tags(volumes=root_volume, vm=virtual_machine_tier1_tag, qos_or_template=self.qos,
+                            disk_offering_id=self.disk_offerings_tier2_tags.id, attached=True)
+        virtual_machine_tier1_tag.stop(self.apiclient, forced=True)
+
+    def test_03_change_offering_on_attached_data_disk(self):
+        virtual_machine_tier1_tag = self.deploy_vm_and_check_tier_tag()
+
+        root_volume = list_volumes(self.apiclient, virtualmachineid=virtual_machine_tier1_tag.id, type="DATADISK",
+                                   listall=True)
+        self.changeOfferingForVolume(root_volume[0].id, self.disk_offerings_tier2_tags.id, root_volume[0].size)
+        root_volume = list_volumes(self.apiclient, virtualmachineid=virtual_machine_tier1_tag.id, type="DATADISK",
+                                   listall=True)
+        self.vc_policy_tags(volumes=root_volume, vm=virtual_machine_tier1_tag, qos_or_template=self.qos,
+                            disk_offering_id=self.disk_offerings_tier2_tags.id, attached=True)
+        virtual_machine_tier1_tag.stop(self.apiclient, forced=True)
+
+    @attr(tags=["advanced", "advancedns", "smoke"], required_hardware="true")
+    def test_04_check_templates_on_deployed_vm_and_datadisk(self):
+        virtual_machine_template_tier1 = VirtualMachine.create(
+            self.apiclient,
+            {"name": "StorPool-%s" % uuid.uuid4()},
+            zoneid=self.zone.id,
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            overridediskofferingid=self.disk_offerings_tier1_template.id,
+            diskofferingid=self.disk_offerings_tier1_template.id,
+            size=2,
+            hypervisor=self.hypervisor,
+            rootdisksize=10
+        )
```

Comment:
```suggestion
        virtual_machine_template_tier1 = VirtualMachine.create(
            self.apiclient,
            {"name": "StorPool-%s" % uuid.uuid4()},
            zoneid=self.zone.id,
            templateid=self.template.id,
            accountid=self.account.name,
            domainid=self.account.domainid,
            serviceofferingid=self.service_offering.id,
            overridediskofferingid=self.disk_offerings_tier1_template.id,
            diskofferingid=self.disk_offerings_tier1_template.id,
            size=2,
            hypervisor=self.hypervisor,
            rootdisksize=10
        )
        self.cleanup.append(virtual_machine_template_tier1)
```

Reply:
Thanks @DaanHoogland, for the review! I'm adding the account to be cleaned and when the cleanup is invoked it destroys everything on that account (VMs, volumes, snapshots, etc.)

```
        cls.account = cls.helper.create_account(
            cls.apiclient,
            cls.services["account"],
            accounttype=1,
            domainid=cls.domain.id,
            roleid=1
        )
        cls._cleanup.append(cls.account)
```

Reply:
ok, no worries or :-1: just remarks